src/main/java/com/google/cloud/resourcemanager/ConstraintInfo.java

```diff
@@ -0,0 +1,301 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.resourcemanager;
+
+import com.google.api.services.cloudresourcemanager.model.BooleanConstraint;
+import com.google.api.services.cloudresourcemanager.model.Constraint;
+import com.google.api.services.cloudresourcemanager.model.ListConstraint;
+import com.google.common.base.Function;
+import com.google.common.base.MoreObjects;
+import java.io.Serializable;
+import java.util.Objects;
+
+/**
+ * A Google Cloud Resource Manager constraint metadata object.
+ *
+ * <p>A Constraint describes a way in which a resource's configuration can be restricted. For
+ * example, it controls which cloud services can be activated across an organization, or whether a
+ * Compute Engine instance can have serial port connections established. Constraints can be
+ * configured by the organization's policy administrator to fit the needs of the organization by
+ * setting Policies for Constraints at different locations in the organization's resource hierarchy.
+ * Policies are inherited down the resource hierarchy from higher levels, but can also be
+ * overridden. For details about the inheritance rules please read about Policies.
+ */
+public class ConstraintInfo implements Serializable {
+
+  private static final long serialVersionUID = 9148970963697734236L;
+
+  static final Function<Constraint, ConstraintInfo> FROM_PROTOBUF_FUNCTION =
+      new Function<Constraint, ConstraintInfo>() {
+        @Override
+        public ConstraintInfo apply(Constraint protobuf) {
+          return ConstraintInfo.fromProtobuf(protobuf);
+        }
+      };
+  static final Function<ConstraintInfo, Constraint> TO_PROTOBUF_FUNCTION =
+      new Function<ConstraintInfo, Constraint>() {
+        @Override
+        public Constraint apply(ConstraintInfo constraintInfo) {
+          return constraintInfo.toProtobuf();
+        }
+      };
+
+  private BooleanConstraint booleanConstraint;
+  private String constraintDefault;
+  private String description;
+  private String displayName;
+  private Constraints constraints;
+  private String name;
+  private Integer version;
+
+  /**
+   * A Constraint that allows or disallows a list of string values, which are configured by an
```

Comment:
you say list but I only see one value below

Reply:
Constraints is nothing but wrapper of `listconstraint `  please check the doc of [listconstraint](https://cloud.google.com/resource-manager/reference/rest/v1/ListAvailableOrgPolicyConstraintsResponse#listconstraint)